### PR TITLE
add draft version of chat with repo compoent

### PR DESCRIPTION
**Pull request by innovation64**
# simple draft version
## function
- chat with generated documents file
- Gradio UI
- local running with basic rag system

## left mission
- chunk code file
- prompt QA
- add vector db to store

### CODE REVIEW

**Comment by Umpire2018** [comment]
Please use [conventionalcommits](https://www.conventionalcommits.org/zh-hans/v1.0.0/)  to write detailed commit next time. You can refer to b396470 .

**Comment by innovation64** [comment]
> Please use [conventionalcommits](https://www.conventionalcommits.org/zh-hans/v1.0.0/) to write detailed commit next time. You can refer to [b396470](https://github.com/LOGIC-10/RepoAgent/commit/b396470501f00a97aacf36120e63e1ce40c6d8b3) .

I'm sorry, I will correct it next time, I just didn' t find the proper way, I i thought just wrote "add draft version of chat with repo component" it's enough

**Comment by LOGIC-10** [comment]
Thank you for contributing! However, please do not commit unverified code directly to the main branch, create a new branch and name it with your new feature, like chat_with_repo. Please submit your changes into this branch

File: repo_agent/chat_with_repo/chat_with_repo.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        chat_history = chat_history + [[message, ""]]
+        auto_merging_response = self.automerging_engine.query(prompt)
+        bot_message = str(auto_merging_response)
+        chat_history.append((message, bot_message))
+        return "", chat_history
+
+# Main Execution Block
+if __name__ == "__main__":
+    openai.api_key = 'sk-'
+    openai.base_url = 'https://sailaoda.cn/v1/'
```
Comment (LOGIC-10) [inline repo_agent/chat_with_repo/chat_with_repo.py:70]: 请不要暴露隐私地址。在提交时用example地址代替

Comment (innovation64) [reply repo_agent/chat_with_repo/chat_with_repo.py:70]: 好的，我马上修改

**Comment by Umpire2018** [comment]
> Thank you for contributing! However, please do not commit unverified code directly to the main branch, create a new branch and name it with your new feature, like chat_with_repo. Please submit your changes into this branch

我认为当前阶段提交到 main 无伤大雅，因为还需要等待审核才能合并，可以先保持到这个状态。

File: repo_agent/chat_with_repo/environment.yml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      - watchfiles==0.21.0
+      - wcwidth==0.2.12
+      - websocket-client==1.7.0
+      - websockets==11.0.3
+      - wheel==0.41.3
+      - win32-setctime==1.1.0
+      - wrapt==1.16.0
+      - yarl==1.9.4
+      - zipp==3.17.0
+prefix: C:\Users\liyan\anaconda3\envs\RepoAgent
```
Comment (Umpire2018) [inline repo_agent/chat_with_repo/environment.yml:203]: 同时，此文件是否不应该提交到 git 上，而是使用更通用的 `requirements.txt` ？

Comment (innovation64) [reply repo_agent/chat_with_repo/environment.yml:203]: 好的

**Pull request closed without merging**